### PR TITLE
add pane primitives

### PR DESCRIPTION
**Pull request by SlicedSilver**
**Type of PR:** enhancement

Update: INCLUDES the Watermark Plugin changes as well.

**PR checklist:**

- [ ] Addresses an existing issue: fixes #
- [x] Includes tests
- [x] Documentation update

**Overview of change:**
Adds Pane Primitives which are attached to a pane instead of requiring a series.

**To do**
- [x] Update documentation guide for Plugins
- [x] Fix interaction test for hit test
- [x] remove attribution and background from graphics e2e test

### CODE REVIEW

File: src/model/pane.ts [hunk 1]
```diff
@@ -354,6 +360,27 @@ export class Pane implements IDestroyable {
 		return this._grid;
 	}
 
+	public attachPrimitive(primitive: IPanePrimitiveBase): void {
+		this._primitives.push(new PanePrimitiveWrapper(primitive));
+	}
+
+	public detachPrimitive(source: IPanePrimitiveBase): void {
```
[the remark this answers is not in the document]
Comment (illetid) [reply src/model/pane.ts:367]: like in cases when we have one series one the pane, then we move this series to another pane, current pane will be destroyed, and if I didn't miss something the primitive's `detached` hook won't be called. I tested with watermark plugin and `detached` hook is not called when we remove pane

[the remark this answers is not in the document]
Comment (SlicedSilver) [reply src/model/pane.ts:367]: Yes, I should add the code to detach any pane primitives when that pane is destroyed. 
Thanks 👍

**Comment by illetid** [comment]
https://github.com/user-attachments/assets/b6413a54-e851-4aee-a083-7bd880705f6c


when I'm attaching the watermark 
```
const imageWatermark = new LightweightCharts.ImageWatermark(imageDataUrl, {
	alpha: 0.5,
	padding: 20,
    maxHeight: 60
});
```
and then resize the pane, the watermark stay at the same position and if I make a pane small enough watermark is not visible completely

**Comment by SlicedSilver** [comment]
> Screen.Recording.2024-08-08.at.13.28.12.mov 
> when I'm attaching the watermark
> 
> ```
> const imageWatermark = new LightweightCharts.ImageWatermark(imageDataUrl, {
> 	alpha: 0.5,
> 	padding: 20,
>     maxHeight: 60
> });
> ```
> 
> and then resize the pane, the watermark stay at the same position and if I make a pane small enough watermark is not visible completely. upd: if we resize chart completely with `chart.resize` position is recalculated

Yeah, that makes sense that is broken. The code is written to do the positioning calculations on the view (assuming there is only one pane). I need to move that placement position logic into the renderer (which has access to the actual pane canvas dimensions). The plugin was written before we had multiple pane support and this wasn't considered.

I'll fix this. Thanks for spotting it.

**Comment by SlicedSilver** [comment]
> @SlicedSilver one minor thing, could you pls check the failing `panes/remove-pane` graphic test which is failing, I tried locally and it works for me, but better to double-check

@illetid 
I've changed the code for the destroying of the pane. Could you please have a look and check?

**Review by illetid — APPROVED** [review]

**Pull request merged**